### PR TITLE
feature: update juicebox strategy to support other protocol versions [juicebox]

### PR DESCRIPTION
**Pull request by jigglyjams**
Changes proposed in this pull request:
- Update juicebox strategy to include `protocolVersion` parameter
- Update examples

### CODE REVIEW

File: src/strategies/juicebox/index.ts [hunk 1]
```diff
@@ -22,7 +26,7 @@ export async function strategy(
     provider,
     abi,
     addresses.map((address: any) => [
-      JBTicketBooth,
+      JBTokenStore[options.protocolVersion],
```
Comment (ChaituVR) [inline src/strategies/juicebox/index.ts:29]:
```suggestion
      options.protocolVersion ? JBTokenStore[options.protocolVersion] : JBTokenStore['1'],
```

**Review by ChaituVR — COMMENTED** [review]
@drgorillamd What you think?

File: src/strategies/juicebox/examples.json [hunk 2]
```diff
@@ -5,14 +5,55 @@
...
+      "0x1e54110DC48dD83F48187b846DF7E32bB76d0D0b"
+    ],
+    "snapshot": 15971323
+  },
+  {
+    "name": "Juicebox Projects",
+    "strategy": {
+      "name": "juicebox",
+      "params": {
+        "projectId": "311",
```
Comment (drgorillamd) [inline src/strategies/juicebox/examples.json:45]: not 1? This might be misleading no?

**Comment by drgorillamd** [comment]
@ChaituVR lgtm

I'm wondering if we shouldn't have a second strat accounting a project across the 3 different versions as one (ie the sum of (v1: idX; v2: idY; v3: idZ)), for project not migrating to one specific release -> need brainstorm, if so, do you rather want it in another PR?
Cheers!

File: src/strategies/juicebox/examples.json (hunk 2 above)
Comment (jigglyjams) [reply src/strategies/juicebox/examples.json:45]: Need to distribute tokens from V3 in order to get the tests to pass

**Review by drgorillamd — APPROVED** [review]

**Comment by jigglyjams** [comment]
@ChaituVR anything else to get this merged in?

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**